pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,128 @@
+# Transforms that generate sysl from arrai based sql model.
+# size returns the size of an attribute.
+let size = \length
+    cond {
+        length = 'MAX': '',
+        //seq.has_prefix('0x', length): $`(${//seq.trim_prefix('0x', length)})`,
+        length > 0: $`(${length})`,
+    };
+
+# hasAttributePattern checks whether an attribute has associated patterns.
+let hasAttributePattern = \entity \attr
+    attr.options?:{} || (entity.primary_key where //seq.contains(attr.name, .)) ||
+    attr.length = 'MAX' || //seq.has_prefix('0x', attr.length) ||
+    ((entity.foreign_keys => ((.foreign_keys where .attribute = attr.name)
+        => .reference_attribute orderby .)) where .);
+
+# hasEntityPattern checks whether an entity has associated patterns.
+let hasEntityPattern = \entity \model
+    entity.primary_key | entity.foreign_keys | entity.cluster | model.indexes;
+
+# sortingOrder determines and appends sorting order.
+let sortingOrder = \e \attr
+    let re = //re.compile($`${attr.name}\((asc|desc)\)`);
+    let keyOrder = e.primary_key => (re.match(.)(0)?(1)?:{} rank (:.@)) where .;
+    cond keyOrder {{x}: $`, ~${x}`};
+
+# compareColumnOrder compares the col order between primary key and table.
+let compareColumnOrder = \entity
+    let pk = (entity.primary_key orderby .) >> //seq.split('(', .) >> .(0);
+    pk !(<=) (entity.attributes >> .name);
+
+# attributePatterns generates the patterns for an attribute.
+let attributePatterns = \entity \attr
+    let options = cond {
+        attr.options:
+            let opt = //seq.split('=',attr.options);
+            $`${opt(0)}='${opt(1)}'`,
+        };
+    let pk = cond {
+        (entity.primary_key where //seq.contains(attr.name, .)):
+            '~pk' ++ sortingOrder(entity, attr)
+    };
+    let fk = cond {
+        (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+        .reference_attribute orderby .)) where .:
+            '~fk'
+    };
+    let length = cond {
+        attr.length = 'MAX': '~max',
+    };
+    let hexPrefix = cond {
+        //seq.has_prefix('0x', attr.length): '~hex',
+    };
+    //seq.join(', ', ([options, pk, fk, length, hexPrefix] where .@item != ''));
```

Comment:
See comment below on return from `entityPatterns` (sorry for being out of order)
```suggestion
    [options, pk, fk, length, hexPrefix] where .@item != '';
```